products-service/src/main/java/com/petclinic/products/presentationlayer/products/ProductController.java

```diff
@@ -68,5 +68,9 @@ public Mono<ResponseEntity<ProductResponseModel>> deleteProduct(@PathVariable St
                 .map(ResponseEntity::ok)
                 .defaultIfEmpty(ResponseEntity.badRequest().build());
     }
+    @GetMapping(value = "/filter/{productType}", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
We implemented GetALL's using a stream value when using flux. Have you tried implemented this?